### DISCUSSION TITLE
Github Copilot LSP

### DISCUSSION THREAD

**Discussion by bkanuka** [python-lsp/python-lsp-server#532] [General]
There seems to be a working Copilot LSP plugin for Sublime Text already written in Python: https://github.com/TerminalFi/LSP-copilot 

Am I thinking clearly that the "hard" parts are already done, and this would be a relatively easy port to a pylsp plugin? 
Would a plugin like this enable Copilot completions in anything that uses python-lsp-server (such as Jupyterlab-lsp https://github.com/jupyter-lsp/jupyterlab-lsp )?

I am _very_ unfamiliar with this, so I would love anyone's contribution or outline of what steps would need to be done for this to work.

**Comment by lieryan** [comment]
If you want to take a stab at creating your own plugin, the [cookiecutter-pylsp-plugin](https://github.com/python-lsp/cookiecutter-pylsp-plugin) might be of interest to you.
